src/controllers/hid/hidcontroller.cpp

```diff
@@ -243,16 +248,40 @@ int HidController::close() {
 bool HidController::poll() {
     Trace hidRead("HidController poll");
 
-    int result = hid_read(m_pHidDevice, m_pPollData, sizeof(m_pPollData) / sizeof(m_pPollData[0]));
-    if (result == -1) {
-        return false;
-    } else if (result > 0) {
+    // This loop risks becoming a high priority endless loop in case processing
+    // the mapping JS code takes longer than the controller polling rate.
+    // This could stall other low priority tasks.
+    // There is no safety net for this because it has not been demonstrated to be
+    // a problem in practice.
+    while (true) {
+        // Cycle between buffers so the memcmp below does not require deep copying to another buffer.
+        unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
+        m_iPollingBufferIndex = (m_iPollingBufferIndex + 1) % kNumBuffers;
+        unsigned char* pCurrentBuffer = m_pPollData[m_iPollingBufferIndex];
+
+        int bytesRead = hid_read(m_pHidDevice, pCurrentBuffer, kBufferSize);
+        if (bytesRead == -1) {
```

Comment:
According to the docs -1 is the only negative value that we need to handle. Nevertheless, I would check for bytesRead < 0 and add a DEBUG_ASSERT(bytesRead == -1) in this if branch. All possible values should be handled.

Reply:
done dff0202